juju/juju
# chore: split provisioner worker in machine and container provisioners

This patch splits the provisioner worker into two new workers: machine and container provisioners.

This follows a previous patch (https://github.com/juju/juju/pull/17954) on which we moved the provisioner task to its own package. The goal of all this work is to facilitate the refactorings and trims on the provisioner which was a huge worker.

Also, as a bonus, I removed the legacy `mocks` sub-package in the machine provisioner in favor new mocks in the root package.

## Checklist


## QA steps

Make sure the unit tests pass:
```
TEST_PACKAGES="./internal/worker/containerprovisioner/... -count=1 -race -check.vv" make run-go-tests
TEST_PACKAGES="./internal/worker/machineprovisioner/... -count=1 -race -check.vv" make run-go-tests
TEST_PACKAGES="./internal/provisionertask/... -count=1 -race -check.vv" make run-go-tests
```

Also make sure that you can launch a container:

```
juju bootstrap aws c
juju add-model m
juju add-machine lxd
```
you should check that the container started correctly when `juju status`.

## Links


**Jira card:** JUJU-6597
